geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -1330,21 +1360,12 @@ private void startMemcachedServer() {
   }
 
   private void startRedisServer() {
-    int port = system.getConfig().getRedisPort();
-    if (port != 0) {
-      String bindAddress = system.getConfig().getRedisBindAddress();
-      assert bindAddress != null;
-      if (bindAddress.equals(DistributionConfig.DEFAULT_REDIS_BIND_ADDRESS)) {
-        getLoggerI18n().info(
-            LocalizedStrings.GemFireCacheImpl_STARTING_GEMFIRE_REDIS_SERVER_ON_PORT_0,
-            new Object[] {port});
-      } else {
-        getLoggerI18n().info(
-            LocalizedStrings.GemFireCacheImpl_STARTING_GEMFIRE_REDIS_SERVER_ON_BIND_ADDRESS_0_PORT_1,
-            new Object[] {bindAddress, port});
-      }
-      this.redisServer = new GeodeRedisServer(bindAddress, port);
-      this.redisServer.start();
+    GeodeRedisService geodeRedisService = getService(GeodeRedisService.class);
```

Comment:
@metatype I suppose the exception call is mostly there because it's the flag for whether Redis is enabled. I think that eventually we would like to have the services be more modular (an "extension framework") and maybe start them without even checking which interfaces they implement, but I don't think that's fully formed yet.